document_parameters.rb

```diff
@@ -1,33 +1,44 @@
-require_relative 'rosette_api_error'
+require_relative 'bad_request_format_error'
 
 class DocumentParameters
   attr_accessor :content, :content_uri, :file_path, :language
 
-  def initialize(content=nil, content_uri=nil, file_path=nil, language=nil)
-    @content = content
-    @content_uri = content_uri
-    @file_path = file_path
-    @language = language
+  def initialize(options = {})
+    options = {
+      'content' => nil,
+      'content_uri' => nil,
+      'file_path' => nil,
+      'language' => nil
+    }.update options
+    @content = options['content']
+    @content_uri = options['content_uri']
+    @file_path = options['file_path']
+    @language = options['language']
   end
 
   def validate_params
-    if [@content, @content_uri, @file_path].count { |attr| !attr.nil? } > 1
-      raise RosetteAPIError.new('badRequestFormat', 'The format of the request is invalid: multiple content sources;' \
-          ' must be one of an attachment, an inline "content" field, or an external "contentUri"')
-    elsif [@content, @content_uri, @file_path].count { |attr| attr.nil? } == 3
-      raise RosetteAPIError.new('badRequestFormat', 'The format of the request is invalid: no content provided; must' \
-          ' be one of an attachment, an inline "content" field, or an external "contentUri"')
+    if [@content, @content_uri, @file_path].compact.length > 1
+      raise BadRequestFormatError.new 'The format of the request is invalid: multiple content sources;' \
+                                      ' must be one of an attachment, an inline "content" field, or an external "contentUri"'
+    elsif [@content, @content_uri, @file_path].all?(&:nil?)
```

Comment:
`#all?` returns `true` if every call to the given block returns `true`.

See http://devdocs.io/ruby~2.2/enumerable#method-i-all-3F for more details.